File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int prefix=1;
        int suffix=1;
        int ans=INT_MIN;
        for(int i=0;i<nums.size();i++){
            if(prefix==0) prefix=1;
            if(suffix==0) suffix=1;
            prefix*=nums[i];
            suffix*=nums[nums.size()-1-i];
            ans=max({ans,prefix,suffix});
        }
        return ans;
        
    }
};

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.py
class Solution:
    def maxProduct(self, nums: list[int]) -> int:
        prefix=1
        suffix=1
        ans=float('-inf')
        for i in range(len(nums)):
            if prefix==0:
                prefix=1
            if suffix==0:
                suffix=1
            prefix*=nums[i]
            suffix*=nums[len(nums)-1-i]
            ans=max({ans,prefix,suffix})
        return ans